src/lib.rs

```diff
@@ -417,6 +493,76 @@ impl<'a> LocalExecutor<'a> {
         task
     }
 
+    /// Spawns many tasks onto the executor.
+    ///
+    /// As opposed to the [`spawn`] method, this locks the executor's inner task lock once and
+    /// spawns all of the tasks in one go. With large amounts of tasks this can improve
+    /// contention.
+    ///
+    /// It is assumed that the iterator provided does not block; blocking iterators can lock up
+    /// the internal mutex and therefore the entire executor. Unlike [`Executor::spawn`], the
+    /// mutex is not released, as there are no other threads that can poll this executor.
+    ///
+    /// ## Example
+    ///
+    /// ```
+    /// use async_executor::LocalExecutor;
+    /// use futures_lite::{stream, prelude::*};
+    /// use std::future::ready;
+    ///
+    /// # futures_lite::future::block_on(async {
+    /// let mut ex = LocalExecutor::new();
+    ///
+    /// let futures = [
+    ///     ready(1),
+    ///     ready(2),
+    ///     ready(3)
+    /// ];
+    ///
+    /// // Spawn all of the futures onto the executor at once.
+    /// let mut tasks = vec![];
+    /// ex.spawn_many(futures, &mut tasks);
+    ///
+    /// // Await all of them.
+    /// let results = ex.run(async move {
+    ///     stream::iter(tasks).then(|x| x).collect::<Vec<_>>().await
+    /// }).await;
+    /// assert_eq!(results, [1, 2, 3]);
+    /// # });
+    /// ```
+    ///
+    /// [`spawn`]: LocalExecutor::spawn
+    /// [`Executor::spawn_many`]: Executor::spawn_many
+    pub fn spawn_many<T: Send + 'a, F: Future<Output = T> + Send + 'a>(
+        &self,
+        futures: impl IntoIterator<Item = F>,
+        handles: &mut impl Extend<Task<F::Output>>,
+    ) {
+        let mut active = self.inner().state().active.lock().unwrap();
```

Comment:
For a local executor, why does it need a lock at all? Probably not necessary to address in this PR, but I thought the local executor itself could be `!Send` and `!Sync`, so wouldn't a RefCell be sufficient for a LocalExecutor state?

Reply:
It would require a reimplementation of `LocalExecutor`'s internals, which is something I don't feel like doing right now as the current implementation works well enough. I would accept a PR for this.

Reply:
ok, as a change this would propagate through the entire code base because it would require a different `State` struct to actually eliminate all the mutexes, etc, which then probably needs either a duplication of all the other internal structs which hold references to `State`, or an abstraction over the two variants of `State`.

Reply:
It's probably doable with `macro_rules`, but that might have a serious impact on readability.

Reply:
I think a proper abstraction would require a lot of interior mutability, we just can't directly abstract all that without pulling in niche dependencies, I suppose..

Reply:
Generally thread unsafe primitives are provided by `unsend` instead of `smol`. In this case see [`unsend::Executor`](https://docs.rs/unsend/latest/unsend/executor/struct.Executor.html)